### DISCUSSION TITLE
hasMany data sync error

### DISCUSSION THREAD

**Discussion by huangzhong0406** [laravel/framework#43166] [Q&A]
- Laravel Version: v9.18.0
- PHP Version: 8.1.7
- Database Driver & Version:  mysql & 5.7.38

### Description:
hi my friend, when I use this code to modify the productDetails data, the result I get is that the productDetails data is created first and then all deleted, I don't understand why, the result I expect is to delete the previous data then create new data

### Steps To Reproduce:
```
    private function handleProductModelAssociate($event)
    {
        $product = $event->getEntity();

        $param = $event->getSubmissionData();

        $description = Arr::get($param, 'description', []);
        $product->productDetails()->delete();

        foreach ($description as $value) {
            $product->productDetails()->create(
                [
                    'product_id' => $product->id,
                    'title' => Arr::get($value, 'title', ''),
                    'description' => htmlspecialchars(Arr::get($value, 'description', ''), ENT_QUOTES)
                ]
            );
        }
    }
```

**Comment by huangzhong0406** [comment]
I can only do this now

![image](https://user-images.githubusercontent.com/29389996/177532453-f6106ed3-32dd-4bc2-94cd-f39d34d929f0.png)

**Comment by huangzhong0406** [comment]
**Can anyone tell me why? I'm confused**

**Comment by yanalshoubaki** [comment]
try to remove the foreach create lines, and keep `        $product->productDetails()->delete();` check if this line remove the data.
if this line not remove check the relation in model, or try to dump the sql query.

this is not a solution just a suggest to know where is the problem.
